lib/PropertyFilter.js

```diff
@@ -1,101 +1,135 @@
+var ERROR_PROPERTY_NOT_FOUND = 'property not found';
+
 var PropertyFilter = function(data) {
     this._data = data;
 };
 
-PropertyFilter.prototype.props = function(props) {
-    if (!props || !props.length) {
-        return this._data;
-    }
-
-    if (this._data === null) {
-        return this._data;
-    }
+var filterProp = function(data, property, filtered) {
+    var filtered = filtered || {};
 
     // Ignore symlinks -- they can't be filtered yet.
-    if (this._data.hasOwnProperty('$link')) {
-        return this._data;
+    // Ignore data that is null
+    if (data === null || data.hasOwnProperty('$link')) {
+        return data;
     }
 
-    if (Array.isArray(this._data)) {
-        props = props.map(function(prop) {
-            return (prop[0] === '@') ? prop.substr(1) : prop;
-        });
-
-        return this._data.map(function(data) {
-            var filter = new PropertyFilter(data);
-            return filter.props(props);
-        });
-    }
-
-    var filtered = {};
-
     // Always keep properties that begin with `$`
-    for (var key in this._data) {
-        if (this._data.hasOwnProperty(key) && key[0] == '$') {
-            filtered[key] = this._data[key];
+    for (var key in data) {
+        if (data.hasOwnProperty(key) && key[0] == '$') {
+            filtered[key] = data[key];
+        }
+    }
+    // Directly assign top-level props or properties beginning with $
+    if (property.indexOf('$') !== -1 || (-1 === property.indexOf('.') && -1 === property.indexOf('@'))) {
```

Comment:
`(-1 === property.indexOf('.') && -1 === property.indexOf('@'))` why do we require that *both* `.` and `@` are in the prop?